### PR TITLE
fix: prevent deletion of 'raster_columns' and 'raster_overviews' PostGIS columns in prisma migrate

### PR DESCRIPTION
**Pull request by jkomyno**
Follow up of [this comment](https://github.com/prisma/prisma/issues/15361) based on [this previous PR](https://github.com/prisma/prisma-engines/pull/3051).

Closes https://github.com/prisma/prisma/issues/15361 and https://github.com/prisma/prisma/issues/15363.

### CODE REVIEW

**Comment by jkomyno** [comment]
(flaky Mac OS compile check, restarted that on Buildkite)

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/sql_schema_differ_flavour/postgres.rs [hunk 1]
```diff
@@ -22,6 +22,8 @@ static POSTGIS_TABLES_OR_VIEWS: Lazy<RegexSet> = Lazy::new(|| {
         "(?i)^spatial_ref_sys$",
         "(?i)^geometry_columns$",
         "(?i)^geography_columns$",
+        "(?i)^raster_columns$",
+        "(?i)^raster_overviews$",
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/sql_schema_differ_flavour/postgres.rs:26]: These tables aren't in the link at the top of the array. Is there a postgis doc page that mentions them? (it does look legit, just that we should document where we got the names)

**Review by tomhoule — COMMENTED** [review]
Looks good! Let's just document where the names came from so we can track in which conditions these tables appear, etc. The list is growing and the risk of false positives is getting larger, we might want to enforce that postgis tables are in a separate schema and then ignore diffing that schema, we could do that with the multi-schema awareness. Or even ignore them conditionally based on the extensions declared in the datasource with the extensions proposal we are implementing. I'll open an issue.

**Comment by tomhoule** [comment]
Here's the issue: https://github.com/prisma/prisma/issues/15363

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/sql_schema_differ_flavour/postgres.rs (hunk 1 above)
Comment (jkomyno) [inline migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/sql_schema_differ_flavour/postgres.rs:26]:
```suggestion
       
        // See §11.2 in: https://postgis.net/docs/using_raster_dataman.html
        "(?i)^raster_columns$",
        "(?i)^raster_overviews$",
```

Comment (jkomyno) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/sql_schema_differ_flavour/postgres.rs:28]: Added a link that mentions those 2 columns

Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/sql_schema_differ_flavour/postgres.rs:28]: Awesome, thanks

**Review by tomhoule — APPROVED** [review]

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/sql_schema_differ_flavour/postgres.rs [hunk 2]
```diff
@@ -22,6 +22,10 @@ static POSTGIS_TABLES_OR_VIEWS: Lazy<RegexSet> = Lazy::new(|| {
         "(?i)^spatial_ref_sys$",
         "(?i)^geometry_columns$",
         "(?i)^geography_columns$",
+       
+        // See §11.2 in: https://postgis.net/docs/using_raster_dataman.html
+        "(?i)^raster_columns$",
+        "(?i)^raster_overviews$",
```
Comment (jkomyno) [inline migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/sql_schema_differ_flavour/postgres.rs:28]:
```suggestion
        // See §11.2 in: https://postgis.net/docs/using_raster_dataman.html
        "(?i)^raster_columns$",
        "(?i)^raster_overviews$",
```

**Pull request merged**